gamemodes/terrortown/gamemode/client/cl_targetid.lua

```diff
@@ -271,19 +271,19 @@ function GM:HUDDrawTargetID()
 		distance = trace.StartPos:Distance(trace.HitPos)
 	end
 
-	ent = hook.Run("TTTModifyTargetedEntity", ent, distance) or ent
-
-	-- make sure it is a valid entity
-	if not IsValid(ent) or ent.NoTarget then return end
-
 	-- if a vehicle, we identify the driver instead
-	if IsValid(ent:GetNWEntity("ttt_driver", nil)) then
+	if IsValid(ent) and IsValid(ent:GetNWEntity("ttt_driver", nil)) then
 		ent = ent:GetNWEntity("ttt_driver", nil)
 	end
 
 	-- only add onscreen infos when the entity isn't the local player
 	if ent == client then return end
```

Comment:
How about checking if it isn't the spectated player as well?

Reply:
The observer target is handles inside the drawcall itself:

https://github.com/TTT-2/TTT2/blob/f5539c32be05e16aa8249f3b0702373a357e4edd/gamemodes/terrortown/gamemode/client/cl_targetid.lua#L761